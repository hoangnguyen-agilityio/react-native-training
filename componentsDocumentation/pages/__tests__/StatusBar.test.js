import React from 'react';
import renderer from 'react-test-renderer';
import StatusBarPage from '../StatusBar';
import {StatusBar, Button} from 'react-native';

test('renders correctly', () => {
  const tree = renderer.create(<StatusBarPage />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('page elements', () => {
  const tree = renderer.create(<StatusBarPage />);

  expect(tree.root.findAllByType(StatusBar)).toHaveLength(1);
});

test('Show / hide StatusBar when click button', () => {
  const tree = renderer.create(<StatusBarPage />);
  const statusBar = tree.root.findAllByType(StatusBar)[0];
  const toggleButton = tree.root.findAllByType(Button)[0];

  expect(statusBar.props.hidden).toBeFalsy();

  toggleButton.props.onPress();

  expect(statusBar.props.hidden).toBeTruthy();

  toggleButton.props.onPress();

  expect(statusBar.props.hidden).toBeFalsy();
});
