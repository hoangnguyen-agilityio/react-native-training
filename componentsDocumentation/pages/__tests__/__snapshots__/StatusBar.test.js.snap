// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#ECF0F1",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "marginBottom": 8,
        "textAlign": "center",
      }
    }
  >
    StatusBar Visibility:
    

    Visible
  </Text>
  <Text
    style={
      Object {
        "marginBottom": 8,
        "textAlign": "center",
      }
    }
  >
    StatusBar Style:
    

    default
  </Text>
  <Text
    style={
      Object {
        "marginBottom": 8,
        "textAlign": "center",
      }
    }
  >
    StatusBar Transition:
    

    fade
  </Text>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Toggle StatusBar
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Change StatusBar Style
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Change StatusBar Transition
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
